src/main/java/core/basesyntax/strategy/BalanceHandler.java

```diff
@@ -0,0 +1,18 @@
+package core.basesyntax.strategy;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.impl.StorageDaoImpl;
+import core.basesyntax.models.FruitTransfer;
+
+public class BalanceHandler implements OperationHandler {
+    private final StorageDao fruitDao;
+
+    public BalanceHandler() {
+        this.fruitDao = new StorageDaoImpl();
+    }
+
+    @Override
+    public void performOperation(FruitTransfer fruitTransfer) {
+        fruitDao.addFruit(fruitTransfer.getFruit(), fruitTransfer.getQuantity());
```

Comment:
What happens if quantity is negative? Let's check it in other handlers too

Reply:
Fixed in all handlers. Any negative value can't be producced.